### PR TITLE
DOC: Updates Index.reindex docstrings (#40328)

### LINKED ISSUE #40328: DOC: pandas Index.reindex
Just noticed that this is poorly documented.

It has signature `(target, method=None, level=None, limit=None, tolerance=None)` and only documents the behaviour of `target`

Is a good candidate to revisit.

### PR DESCRIPTION
**Pull request by lnros**
- [x] closes #40328: explains function parameters and gives basic example
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see here for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -3763,18 +3763,59 @@ def _validate_can_reindex(self, indexer: np.ndarray) -> None:
 
     def reindex(self, target, method=None, level=None, limit=None, tolerance=None):
         """
-        Create index with target's values.
+        Create an index with target's values.
 
         Parameters
         ----------
         target : an iterable
+        method : {None, ‘backfill’/’bfill’, ‘pad’/’ffill’, ‘nearest’}
```
Comment (phofl) [inline pandas/core/indexes/base.py:3771]: Is there a way to share parts with NDFrame.reindex?

Comment (lnros) [reply pandas/core/indexes/base.py:3771]: Maybe share parts with other functions in this same file, such as get_indexer? @phofl

**Review by datapythonista — COMMENTED** [review]
Thanks for the contribution @lnros, and sorry for the delay in reviewing. Great contribution, added some small comments about the standards we use, but looks great. Ping me when you address the comments, and the CI is green, and I'll have another look.

**Comment by simonjayhawkins** [comment]
@lnros can you merge master to resolve conflicts

**Comment by lnros** [comment]
@simonjayhawkins Done

**Comment by jreback** [comment]
closing as stale

**Pull request closed without merging**